src/FieldEncryptProcessEntities.php

```diff
@@ -138,16 +120,14 @@ protected function process_field(\Drupal\Core\Field\FieldItemListInterface $fiel
 
     $field_type = $definition->get('field_type');
 
-    /**
-     * Filter out fields that do not have a defined map.
-     */
-    if (!in_array($field_type, array_keys($this->getFieldEncryptMap()))) {
+    // Filter out fields that do not have a defined encryption provider for
+    // their field type.
+    $supported_types = $this->providerPluginManager->getSupportedFieldTypes();
```

Comment:
Yes, I think this is limiting. See above comments. I think we check _all_ fields which field encrypt setting is checked and we have a general purpose utility to select properties of the field to encrypt.